pkg/export/otel/otelcfg/config_metrics.go

```diff
@@ -121,7 +121,7 @@ func (m *MetricsConfig) EndpointEnabled() bool {
 }
 
 func (m *MetricsConfig) AnySpanMetricsEnabled() bool {
-	return m.SpanMetricsEnabled() || m.SpanMetricsSizesEnabled()
+	return m.SpanMetricsEnabled() || m.SpanMetricsSizesEnabled() || m.ServiceGraphMetricsEnabled()
```

Comment:
This was recently introduced bug with the refactoring to make spangraphs generated separately. I've added a test now to ensure this doesn't happen anymore.